crates/net/rpc/src/engine/mod.rs

```diff
@@ -16,28 +17,122 @@ use reth_rpc_types::engine::{
 mod error;
 pub use error::EngineApiError;
 
-/// TODO:
-pub struct EngineApi {
+/// The Engine API result type
+pub type EngineApiResult<Ok> = std::result::Result<Ok, EngineApiError>;
+
+/// The server implementation of Engine API
+pub struct EngineApi<Eth> {
     /// The implementation of `eth` API
-    eth: Box<dyn EthApiSpec>,
+    eth: Eth,
     /// Consensus configuration
     config: Config,
 }
 
-impl std::fmt::Debug for EngineApi {
+impl<Eth> std::fmt::Debug for EngineApi<Eth> {
     fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
-        f.debug_struct("EngineApi").finish_non_exhaustive()
+        f.debug_struct("EngineApi").field("config", &self.config).finish_non_exhaustive()
+    }
+}
+
+impl<Eth> EngineApi<Eth>
+where
+    Eth: EthApiSpec + 'static,
```

Comment:
This will be used not only in RPC, right?

imo this makes it more verbose and is not worth it. but can just leave it and reevaluate later.